omod_1.x/src/main/java/org/openmrs/module/webservices/rest/resource/ItemStockDetailBaseResource.java

```diff
@@ -47,4 +48,12 @@ public void setExpiration(ItemStockDetailBase instance, String dateText) {
 
 		instance.setExpiration(date);
 	}
+
+	@PropertySetter("batchNumber")
+	public void setBatchNumber(ItemStockDetailBase instance, String batchNumber) {
+		if (batchNumber == null) {
```

Comment:
If batchNumber is null or batchNumber.trim().isEmpty() -- throw error.

Reply:
@gsluthra updated this as suggested please review.